src/js/plugins/menu_toggle/plugin.menu_toggle.js

```diff
@@ -1,113 +1,194 @@
+        || $(event.target).hasClass('BRpageview'); /* empty page placeholder */
+    };
 
-      var brContainer = document.querySelector('.BRcontainer') || {};
-      var mainBRWrapper = document.querySelector('.BookReader') || {};
-      var firstChild = brContainer.firstChild;
-  
-      if (firstChild) {
-        firstChild.addEventListener('click', function(e) {
-          toggleMenuIfCenterClick(br, e);
-        }, true);
-      }
+    /**
+     * Main hook into toggle functionality
+     * This is the only function that should be called by the event handlers
+     *
+     * @param { object } br - BookReader instance
+     * @param { object } e - JS event object
+     * @param { boolean } atBookCenter - optional
+     */
+    var toggleRouter = function toggleRouter (br, e, atBookCenter) {
+      var isValidClickArea = atBookCenter ? isCenterClick(e) : isBackground(e);
 
-      var toggleAtBackgroundClick = function toggleAtBackgroundClick(e) {
-        var isBackground = $(event.target).hasClass('BookReader')
-          || $(event.target).hasClass('BRcontainer') /* main black theatre */
-          || $(event.target).hasClass('BRemptypage') /* empty page placeholder */
-          || $(event.target).hasClass('BRpageview'); /* empty page placeholder */
-        if (isBackground) {
-          toggleNav(br);
+      if (isValidClickArea) {
+        toggleNav(br, atBookCenter);
+
+        if (atBookCenter) {
+          e.stopPropagation(); // don't turn the page. this takes prescendence
         }
-      };
-      mainBRWrapper.addEventListener('click', toggleAtBackgroundClick, true);
+      }
+    }
+
+    /**
+     * background click event handler
+     * @param { object } br - BookReader instance
+     * @param { object } e - JS event object
+     */
+    function onBackgroundClick(br, e) {
+      toggleRouter(br, e);
+    }
+
+    /**
+     * actual book container click event handler
+     *
+     * @param { object } br - BookReader instance
+     * @param { object } e - JS event object
+     */
+    function onBookClick(br, e) {
+
+      var atBookCenter = true;
+      toggleRouter(br, e, atBookCenter);
     }
 
-    BookReader.prototype.initMenuToggle = function brInitMenuToggle(e) {
+    /**
+     * attaches click handlers to background & book
+     * @param { object } br - BookReader instance
+     */
+    function registerClickHandlers(br) {
+      var background = document.querySelector('.BookReader') || {};
+      background.addEventListener('click', onBackgroundClick.bind(null, br), { capture: true, passive: true });
+
+      var desk = document.querySelector('.BRcontainer') || {};
+      var book = desk.firstChild;
+
+      if (book) {
+        book.addEventListener('click', onBookClick.bind(null, br), true);
+      }
+    }
+
+    /**
+     * Install menu toggle
+     * attaches event handlers, sets up DOM on load
+     */
+    BookReader.prototype.install = function installMenuToggle(e) {
```

Comment:
I might name this method more specifically.. `install` is kind of generic and makes me think it's installing the entire book reader, but it's just installing the menu toggle right?

Reply:
yes, +1 to more explicit